src/xdist/newhooks.py

```diff
@@ -55,3 +55,13 @@ def pytest_xdist_node_collection_finished(node, ids):
 @pytest.mark.firstresult
 def pytest_xdist_make_scheduler(config, log):
     """ return a node scheduler implementation """
+
+
+@pytest.mark.firstresult
+def pytest_xdist_auto_num_workers(config):
```

Comment:
my personal preference would be not to have a hook, and just fall back to the less accurate mechanism if psutil is missing

Reply:
I went with a hook because it also allows for further customization without having to change the core. What do you think?

Reply:
my first impulse is YAGNI

Reply:
but lets see what users will do